credentials/alts/internal/conn/record.go

```diff
@@ -185,17 +212,27 @@ func (p *conn) Read(b []byte) (n int, err error) {
 		}
 		ciphertext := msg[msgTypeFieldSize:]
 
-		// Decrypt requires that if the dst and ciphertext alias, they
```

Comment:
Can you explain why we get to ignore this now?

Reply:
I changed this to use a separate destination buffer for decryption, so the aliasing requirement mentioned here was no longer applicable. I've reverted this change and restored the previous behaviour of using the input slice as the destination as there's no performance benefit of using a separate destination slice.

I introduced the new destination buffer to hold multiple decrypted frames, but that didn't help so I reverted it.